File: Arrays/P001.py
def brute(arr):
    n=len(arr)
    for i in range(n):
        c=0
        for j in range(n):
            if arr[j]>arr[i]:
                c=1
                break
        if c==0:
            return arr[i]
    return -1

def optimal(arr):
    n=len(arr)
    max_ele=arr[0]
    for i in range(1,n):
        if arr[i]>max_ele:
            max_ele=arr[i]
    return max_ele

if __name__=='__main__':
    arr=input('Enter elements')
    arr=arr.split()
    print('Brute: ',brute(arr))
    print('Optimal: ',optimal(arr))

File: Stiver_Problems/Arrays/P001.cpp
// P001.cpp - Find maximum element (brute and optimal)

#include <bits/stdc++.h>
using namespace std;

int brute(const vector<int>& arr) {
    int n = arr.size();
    for (int i = 0; i < n; i++) {
        bool bigger = false;
        for (int j = 0; j < n; j++) {
            if (arr[j] > arr[i]) {
                bigger = true;
                break;
            }
        }
        if (!bigger) return arr[i];
    }
    return -1;
}

int optimal(const vector<int>& arr) {
    if (arr.empty()) return -1;
    int max_ele = arr[0];
    for (int x : arr) {
        if (x > max_ele) max_ele = x;
    }
    return max_ele;
}

int main() {
    cout << "Enter elements: ";
    string line;
    getline(cin, line);
    stringstream ss(line);
    vector<int> arr;
    int x;
    while (ss >> x) arr.push_back(x);

    cout << "Brute: " << brute(arr) << "\n";
    cout << "Optimal: " << optimal(arr) << "\n";
    return 0;
}